pywbem/_subscription_manager.py

```diff
@@ -1253,47 +1268,92 @@ def _create_filter(self, server_id, source_namespace, query,
         # Validate server_id
         server = self._get_server(server_id)
 
-        this_host = getfqdn()
         ownership = "owned" if owned else "permanent"
 
-        filter_path = CIMInstanceName(FILTER_CLASSNAME,
-                                      namespace=server.interop_ns)
+        this_client = getfqdn()
 
         filter_inst = CIMInstance(FILTER_CLASSNAME)
-        filter_inst.path = filter_path
         filter_inst['CreationClassName'] = FILTER_CLASSNAME
         filter_inst['SystemCreationClassName'] = SYSTEM_CREATION_CLASSNAME
-        filter_inst['SystemName'] = this_host
+        filter_inst['SystemName'] = self._systemnames[server_id]
         if filter_id:
             filter_inst['Name'] = _format(
-                'pywbemfilter:{0}:{1}:{2}:{3}',
-                ownership, self._subscription_manager_id, filter_id,
-                uuid.uuid4())
+                'pywbemfilter:{0}:{1}:{2}:{3}:{4}',
+                ownership, this_client, self._subscription_manager_id,
+                filter_id, uuid.uuid4())
         if name:
             filter_inst['Name'] = name
+
         filter_inst['SourceNamespace'] = source_namespace
         filter_inst['Query'] = query
         filter_inst['QueryLanguage'] = query_language
 
         if owned:
```

Comment:
The two paths for owned vs permanent can now become one path, i.e. remove the "if" and the "then" block and make the "else" block the remaining code.